### ISSUE TITLE
Input mode bug

### ISSUE THREAD

**Issue by karljs** [banacorn/agda-mode#2]
Let me start by saying that I am *very* appreciative of your work on this.  Although I don't hate Emacs, I look forward to the day when people actually have a choice of editor when it comes to working with Agda code.  As for the bug...

Try the following:
1. Open an agda file and load it (why is loading necessary to enable the input mode, anyway?)
2. Type "\to" to insert an arrow, but do not type space afterward.
3. Type backspace (doesn't delete arrow)
4. Type any other character (produces a strange second arrow)

I'd really like to work contribute a patch, but I haven't spent the time to figure out Atom's plugin architecture yet.

**Comment by banacorn** [comment]
Hey, thanks for trying this out :D 

I've successfully reproduced the issue you mentioned (actually I bumped into these ~5 times a day, but there are many other bugs that are more annoying than this one, so ...... I just shut Atom and play video games haha :p ).

Anyway, the patch will be delivered soon!

By the way, loading was not necessary to enable the input mode before, but I opted it out for no reason when I rewrote the whole codebase when it comes to  `v0.2.x`. Yes, I should put it back.

The Atom API is still a mess, but [1.0](https://discuss.atom.io/t/atom-roadmap-to-1-0/12379) is coming.
All patches are welcome! and thanks again!
